modules/gapi/perf/render/gapi_render_perf_tests_inl.hpp

```diff
@@ -0,0 +1,1018 @@
+            {
+                cell_roi = msc_roi(cv::Rect(cols, i, msc_roi.cols - cols, cellSz));
+                cell_roi = cv::mean(cell_roi);
+            }
+        }
+
+        if (has_crop_y)
+        {
+            for(int j = 0; j < cols; j += cellSz)
+            {
+                cell_roi = msc_roi(cv::Rect(j, rows, cellSz, msc_roi.rows - rows));
+                cell_roi = cv::mean(cell_roi);
+            }
+            if (has_crop_x)
+            {
+                cell_roi = msc_roi(cv::Rect(cols, rows, msc_roi.cols - cols, msc_roi.rows - rows));
+                cell_roi = cv::mean(cell_roi);
+            }
+        }
+    }
+
+    // TODO: remove this code duplication with /gapi/test/common/gapi_render_tests.hpp
+    void blendImageRef(cv::Mat& mat, const cv::Point& org, const cv::Mat& img, const cv::Mat& alpha)
+    {
+        auto roi = mat(cv::Rect(org, img.size()));
+        cv::Mat img32f_w;
+        cv::merge(std::vector<cv::Mat>(3, alpha), img32f_w);
+
+        cv::Mat roi32f_w(roi.size(), CV_32FC3, cv::Scalar::all(1.0));
+        roi32f_w -= img32f_w;
+
+        cv::Mat img32f, roi32f;
+        img.convertTo(img32f, CV_32F, 1.0/255);
+        roi.convertTo(roi32f, CV_32F, 1.0/255);
+
+        cv::multiply(img32f, img32f_w, img32f);
+        cv::multiply(roi32f, roi32f_w, roi32f);
+        roi32f += img32f;
+
+        roi32f.convertTo(roi, CV_8U, 255.0);
+    }
+
+  }
+
+  INSTANTIATE_TEST_CASE_P(RenderTestTexts, RenderTestTexts,
+      Combine(Values(std::string("Some text")),
+              Values(szVGA, sz720p, sz1080p),
+              Values(AbsExact().to_compare_f())));
+
+  INSTANTIATE_TEST_CASE_P(RenderTestRects, RenderTestRects,
+      Combine(Values(szVGA, sz720p, sz1080p),
+              Values(AbsExact().to_compare_f())));
+
+  INSTANTIATE_TEST_CASE_P(RenderTestCircles, RenderTestCircles,
+      Combine(Values(szVGA, sz720p, sz1080p),
+              Values(AbsExact().to_compare_f())));
+
+  INSTANTIATE_TEST_CASE_P(RenderTestLines, RenderTestCircles,
+      Combine(Values(szVGA, sz720p, sz1080p),
+              Values(AbsExact().to_compare_f())));
+
+  INSTANTIATE_TEST_CASE_P(RenderTestMosaics, RenderTestCircles,
+      Combine(Values(szVGA, sz720p, sz1080p),
+              Values(AbsExact().to_compare_f())));
+
+  INSTANTIATE_TEST_CASE_P(RenderTestImages, RenderTestCircles,
+      Combine(Values(szVGA, sz720p, sz1080p),
+              Values(AbsExact().to_compare_f())));
+
+  INSTANTIATE_TEST_CASE_P(RenderTestPolylines, RenderTestPolylines,
+      Combine(Values(szVGA, sz720p, sz1080p),
+              Values(AbsExact().to_compare_f())));
+
+  INSTANTIATE_TEST_CASE_P(RenderTestPolyItems, RenderTestPolyItems,
+      Combine(Values(szVGA, sz720p, sz1080p),
+              Values(50),
+              Values(50),
+              Values(50),
+              Values(AbsExact().to_compare_f())));
+
+PERF_TEST_P_(RenderTestTexts, RenderTextsPerformanceBGROCVTest)
+{
+    cv::Point org(200, 200);
```

Comment:
All parameters should be as tests parameters

Reply:
Fixed